src/Nethermind/Nethermind.Init/Steps/InitializeNetwork.cs

```diff
@@ -307,20 +339,25 @@ private void InitDiscovery()
         IDiscoveryConfig discoveryConfig = _api.Config<IDiscoveryConfig>();
 
         SameKeyGenerator privateKeyProvider = new(_api.NodeKey.Unprotect());
-        NodeIdResolver nodeIdResolver = new(_api.EthereumEcdsa);
 
-        if (discoveryConfig.Discv5Enabled)
+        _api.DiscoveryConnections = new DiscoveryConnectionsPool(_logger, _networkConfig, discoveryConfig);
+
+        // TODO should port checks or separate bool values be used here?
```

Comment:
This needs to be discussed

Reply:
https://github.com/NethermindEth/nethermind/pull/7213/files#diff-bd19098570ab009a94a742e85abdcecd40471f86ef75fda621acbef5408b1bc0R353

Reply:
Question is - how should we specify these options in config:
- V4 enabled/disabled
- V4 UDP port
- V5 enabled/disabled
- V5 UDP port

For now I'm using `0` or negative port values as a "protocol disabled" configuration.

Reply:
V4 uses `DiscoveryEnabled` from `IInitConfig`. Should we move v5 there also?

Reply:
`DiscoveryEnabled` is whether discovery runs at all, and IDiscoveryConfig may tell which version is used. Hm maybe just add `Version` = "discv4/discv5/both" or something like that.

And yeah it might be better instead of Init.DiscoveryEnabled to have Discovery.Enabled, but it is how it is. We can make any flag obsolete but no sure if it's worth it